crates/forge/src/collecting.rs

```diff
@@ -5,13 +5,18 @@ use assert_fs::TempDir;
 use cairo_lang_sierra::program::Program;
 use camino::{Utf8Path, Utf8PathBuf};
 use rayon::prelude::{IntoParallelRefIterator, ParallelIterator};
-use test_collector::{collect_tests, LinkedLibrary, TestCase};
+use test_collector::{
+    collect_tests, ForkConfig, LinkedLibrary, RawForkConfig, TestCase, ValidatedForkConfig,
+};
 use walkdir::WalkDir;
 
+pub(crate) type CompiledTestCrateRaw = CompiledTestCrate<RawForkConfig>;
+pub(crate) type CompiledTestCrateRunnable = CompiledTestCrate<ValidatedForkConfig>;
+
 #[derive(Debug, Clone)]
-pub(crate) struct CompiledTestCrate {
+pub(crate) struct CompiledTestCrate<T: ForkConfig> {
     pub sierra_program: Program,
-    pub test_cases: Vec<TestCase>,
+    pub test_cases: Vec<TestCase<T>>,
```

Comment:
Let's also add a type `TestCaseRunnable` or `RunnableTestCase` that is `TestCase<CompiledTestCrateRunnable>` and use it in methods in `lib.rs` and `running.rs`.